src/query/api/v1/handler/prom/read.go

```diff
@@ -177,6 +251,27 @@ func (h *readHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Rulemanager results are vector values (list of metric + value)
+	// Take a random number and check if under rate so we check a proportion of the queries
+	if rand.Float64() < float64(h.queryCheckConfig.CheckSampleRate) && res.Value.Type() == parser.ValueTypeVector {
+		query, err := h.opts.newQueryFn(params)
+		if err != nil {
+			h.logger.Error("Comparison query failed to create")
+		}
+		defer query.Close()
+		// Set context so we can default to M3DB later on
+		result := query.Exec(context.WithValue(ctx, "UseM3DB", true))
+		if result.Err != nil {
+			h.logger.Error("Comparison query failed to execute")
+		} else {
+			if result != nil && !compareResults(res, result, h.queryCheckConfig.ComparePercentThreshold) {
```

Comment:
How do you know this `res` if queried from cache instead of m3db if cache miss the hit?